src/modes/map/map_mode.cpp

```diff
@@ -1073,6 +1073,19 @@ void MapMode::_DrawMapLayers()
 {
     VideoManager->PushState();
     VideoManager->SetStandardCoordSys();
+
```

Comment:
Somewhere in here is where I imagine all the heavy lifting will eventually happen.

Ideally, I think everything (Not just map tiles; the map objects and the player character too.) should be drawn at the native scale of the map tiles which I think is 32x32 pixels.  Then, scaled up at the end.

I did some initial tinkering to try and diagnose what would need to happen.  However, I ran into some issues.  I could never get the map tiles and all the map objects to align in any other way than they are currently.  Swapping their coordinate systems and scales around seemed pretty tough to do correctly.